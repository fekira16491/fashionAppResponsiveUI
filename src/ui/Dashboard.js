import React, { useState } from "react";
import Jane from "../fashionImages/jane.jpg";
import GirlHat from "../fashionImages/girl_hat.jpg";
import Avatar from "../fashionImages/avatar_girl2.jpg";
import AvatarSmoke from "../fashionImages/avatar_smoke.jpg";
import { useSelector, useDispatch } from 'react-redux'
import { decrement, increment } from  "../redux/store/slice/counterSlice"

const Dashboard = () => {
  const [count, setCount] = useState(0);

  const count1 = useSelector((state) => state.counter.value)
  const dispatch = useDispatch()

  const [timeListValue,settimeListValue] = useState("")
  const input = [
    { id: 1, url: "https://www.url1.dev" },
    { id: 2, url: "https://www.url2.dev" },
    { id: 3, url: "https://www.link3.dev" },
  ];

  const filterByTerm = () => {
    var obj = { id: 3, url: "https://www.link3.dev" };
    console.log("object", obj);
    setCount(count + 1);
    return obj;
  };
  const handlechange = (e) => {
  settimeListValue(e.target.value)
  };

  return (
    <div>
      <header>
        <div className="headerSection">
          <div className="headerLeftSection">
            <label>Insta</label>
            <label>Facebook</label>
            <label>Whatsapp</label>
            <label>{count1}</label>
            <select id="dropdown" onChange={handlechange} value={timeListValue}>
              {input.map((item) => {
                return (
                  <option key={item.id} value={item.id}>
                    {item.url}
                  </option>
                );
              })}
            </select>{" "}
          </div>
          <div>
            <label>Login</label>
          </div>
        </div>
      </header>
      <body>
        <div>
          <div className="displayFlexColumn kingHeader">
            <h1>King BLOGLIFE</h1>
            <h4>Welcome to the blog of janes world.</h4>
          </div>

          <div className="divRelative">
            <img src={Jane} alt="jane" />
            <div className="divAbsolute">
              <h3 style={{ marginBottom: "1rem" }}>King's </h3>
              <h2>Fashion Blog</h2>
              <button id="filterButton" onClick={()=>{dispatch(increment())}}>
                Subscribe
              </button>
            </div>
          </div>

          <div className="displayFlexMediaQuery subSectionDesign">
            <div className="displayTitleBox">
              <div className="displayFlexColumn justContentAndAlignCenter">
                <h6 data-testid="title">Title Heading</h6>

                <h6>Title Description,May 2,2016.</h6>
              </div>

              <img src={GirlHat} alt="hat"></img>
              <p>
                More Hats! I am crazy about hats these days. Some text about
                this blog entry. Fashion fashion and mauris neque quam,
                fermentum ut nisl vitae, convallis maximus nisl. Sed mattis this
                blog entry. Fashion fashion and mauris neque quam, fermentum ut
                nisl vitae, convallis maximus nisl. Sed mattis
              </p>
              <p>
                More Hats! I am crazy about hats these days. Some text about
                this blog entry. Fashion fashion and mauris neque quam,
                fermentum ut nisl vitae, convallis maximus nisl. Sed mattis this
                blog entry. Fashion fashion and mauris neque quam, fermentum ut
                nisl vitae, convallis maximus nisl. Sed mattis
              </p>
            </div>
            <div className="displayFlexWith3columns ">
              <div className="displayFlexColumn">
                <img src={Avatar} alt="hat"></img>
                <div className="titleName">
                  <h5>My Name</h5>
                  <h5>
                    Just me, myself and I, exploring the universe of uknownment.
                    I have a heart of love and a interest of lorem ipsum and
                    mauris neque quam blog. I want to share my world with you.
                  </h5>
                </div>
              </div>

              <div
                className="displayFlexColumn"
                style={{ paddingTop: "1rem", margin: 0, width: "100%" }}
              >
                <div className="titleName">
                  <h5>My Popular post</h5>
                </div>
                <div class="displayFlex" style={{ padding: 5 }}>
                  <div class="image">
                    <img
                      src={AvatarSmoke}
                      alt="avatar"
                      width="70"
                      height="70"
                    />
                  </div>
                  <div
                    className="displayFlexColumn"
                    style={{
                      marginLeft: "1em",
                      justifyContent: "space-evenly",
                    }}
                  >
                    <h5 style={{ margin: 0 }}>Denim</h5>
                    <h5 style={{ margin: 0 }}>Sed mattis nunc</h5>
                  </div>
                </div>

                <hr className="solid" />

                <div class="displayFlex" style={{ padding: 5 }}>
                  <div class="image">
                    <img
                      src={AvatarSmoke}
                      alt="avatar"
                      width="70"
                      height="70"
                    />
                  </div>
                  <div
                    className="displayFlexColumn"
                    style={{
                      marginLeft: "1em",
                      justifyContent: "space-evenly",
                    }}
                  >
                    <h5 style={{ margin: 0 }}>Denim</h5>
                    <h5 style={{ margin: 0 }}>Sed mattis nunc</h5>
                  </div>
                </div>

                <hr className="solid" />

                <div class="displayFlex" style={{ padding: 5 }}>
                  <div class="image">
                    <img
                      src={AvatarSmoke}
                      alt="avatar"
                      width="70"
                      height="70"
                    />
                  </div>
                  <div
                    className="displayFlexColumn"
                    style={{
                      marginLeft: "1em",
                      justifyContent: "space-evenly",
                    }}
                  >
                    <h5 style={{ margin: 0 }}>Denim</h5>
                    <h5 style={{ margin: 0 }}>Sed mattis nunc</h5>
                  </div>
                </div>

                <hr className="solid" />
              </div>

              <div
                className="displayFlexColumn"
                style={{
                  paddingTop: "1rem",
                  margin: 0,
                  width: "100%",
                  height: "100%",
                }}
              >
                <div className="titleName">
                  <h5>My Popular post</h5>
                </div>
                <div
                  className="box"
                  style={{
                    height: "100%",
                    width: "auto",
                    backgroundColor: "grey",
                    border: "15px solid #f1f1f1",
                    padding: "30%",
                    margin: "0px",
                  }}
                >
                  <label style={{ display: "flex", justifyContent: "center" }}>
                    Your ad here.
                  </label>
                  {/* <div className="centerBox" style={{width:"80%" ,height:"80%",padding:"3rem" , backgroundColor:"red",margin:"0px" }}>
</div> */}
                </div>
              </div>
            </div>
          </div>
        </div>
      </body>
    </div>
  );
};

export default Dashboard;
