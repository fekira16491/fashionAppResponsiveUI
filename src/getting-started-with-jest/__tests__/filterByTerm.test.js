import Dashboard from "../../ui/Dashboard";
import * as dash from "../../ui/Dashboard";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { mount, shallow } from "enzyme";

let wrapper;
beforeEach(() => {
  wrapper = mount(<Dashboard />);
});

it("should render Dashboard comp", () => {
  expect(wrapper).toHaveLength(1);
});
describe("Given Dashboard comp", () => {
  // it('should take a snapshot', () => {

  //     // eslint-disable-next-line testing-library/no-debugging-utils
  //       const utils = render(<Dashboard />);

  //      // eslint-disable-next-line testing-library/no-debugging-utils
  //      console.log("wrapper",wrapper.debug());
  //      // eslint-disable-next-line testing-library/prefer-screen-queries
  //      // eslint-disable-next-line no-undef
  //     expect(wrapper).toHaveLength(1);
  //      // eslint-disable-next-line testing-library/prefer-screen-queries
  //      const filterButton = utils.getByTestId('filterButton').find('button').find('filterButton');;
  //      expect(filterButton).toEqual('Subscribe');

  //     //const incrementBtn = wrapper.find('filterButton');

  //     //expect(incrementBtn).();

  //     // incrementBtn.simulate('click');

  //      //expect(hello).toHaveBeenCalled();
  //     //const FakeFun = jest.spyOn(dash, 'filterByTerm');

  //     // const output = { id: 3, url: "https://www.link3.dev" };;

  //     // var jestFunc = jest.fn();
  //     // const incrementBtn = wrapper.find('button');
  //     // incrementBtn.simulate('click');
  //     // const filterButton = wrapper.find('button').find('#filterByTerm').
  //     // find('button');
  //     // filterButton.simulate('click');
  //     // expect(jestFunc).toHaveBeenCalled();
  //    // expect(FakeFun).toHaveBeenCalled();

  //     ///expect(text).toEqual('Subscribe');
  //   //  expect(incrementBtn).toHaveBeenCalledWith(jestFunc);
  //     //xpect(jestFunc).toHaveBeenCalled();

  // //     const utils = render(<Dashboard />);

  // //     const container = utils.getByTestId('title');
  // //     const container1 = utils.getByTestId("button')
  // //     expect(utils.getByText('Title Heading', container)).toBeTruthy();
  // // expect(container).
  // //     expect(onClick).toHaveBeenCalledTimes(1);

  //     // const wrapper = shallow(<Dashboard />);
  //     // const text = wrapper.find('p').text();

  // })
  // })
  // describe("filterByTerm",()=>{
  //     it("should return filterByTerm action obj",()=>{
  //     const input = [
  //         {id:1,url: "https://www.url1.dev"},
  //         {id:2,url:"https://www.url2.dev"},
  //         {id:3,url:"https://www.link3.dev"}
  //     ]
  //     const output = [{ id: 3, url: "https://www.link3.dev" }];

  //     const wrapper = mount(<Dashboard/>);
  //     const submitButton = wrapper.find('button');
  //     submitButton.simulate('click');
  //     expect(submitButton).toEqual(output);
  // })

  describe('when "filter button" is clicked', () => {
    it("should  call filter button click", () => {
        var obj ={id: 3, url: "https://www.link3.dev"} 

      const buttonElement = wrapper.find("#filterButton");
      expect(buttonElement.text()).toEqual("Subscribe");
      buttonElement.simulate("click");
    //   const text = wrapper.find('#filterCount');
    //   expect(text.text()).toEqual("1");
     let item = wrapper.find('option').at(0).instance().selected = false;
      wrapper.find('option').at(1).instance().selected = true;
      wrapper.find('select').simulate('change', {target: {value: 'https://www.url1.dev'}})

  // Then its value changes to 10:00
  expect(wrapper.find('select').props().value).toBe('https://www.url1.dev')
    //   expect(wrapper.find('select').props().value).toBe('10:00')

    // console.log("item",item)
      // expect(buttonElement).toHaveLength(1);

      //   render(<Dashboard />);
      //   const handleClick = jest.fn();
      //   wrapper
      // .find('button#filterButton')
      // .simulate('onClick',{keyCode: 27});
      //   const filterButton = screen.getByTestId("filterButton");
      //   const filterText = screen.getByText('Subscribe');
      //   console.log("filterText",filterText)
      //fireEvent.click(screen.getByText('Subscribe'))
      //expect(handleClick).toHaveBeenCalledTimes(1);
    });
  });
});
