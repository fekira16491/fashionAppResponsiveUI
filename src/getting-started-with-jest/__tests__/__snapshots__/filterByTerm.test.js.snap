// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <header>
      <div
        class="headerSection"
      >
        <div
          class="headerLeftSection"
        >
          <label>
            Insta
          </label>
          <label>
            Facebook
          </label>
          <label>
            Whatsapp
          </label>
          <label>
            Twitter
          </label>
        </div>
        <div>
          <label>
            Login
          </label>
        </div>
      </div>
    </header>
    <div>
      <div
        class="displayFlexColumn kingHeader"
      >
        <h1>
          King BLOGLIFE
        </h1>
        <h4>
          Welcome to the blog of janes world.
        </h4>
      </div>
      <div
        class="divRelative"
      >
        <img
          alt="jane"
          src="jane.jpg"
        />
        <div
          class="divAbsolute"
        >
          <h3
            style="margin-bottom: 1rem;"
          >
            King's 
          </h3>
          <h2>
            Fashion Blog
          </h2>
          <button>
            Subscribe
          </button>
        </div>
      </div>
      <div
        class="displayFlexMediaQuery subSectionDesign"
      >
        <div
          class="displayTitleBox"
        >
          <div
            class="displayFlexColumn justContentAndAlignCenter"
          >
            <h6>
              Title Heading
            </h6>
            <h7>
              Title Description,May 2,2016.
            </h7>
          </div>
          <img
            alt="hat"
            src="girl_hat.jpg"
          />
          <p>
            More Hats! I am crazy about hats these days. Some text about this blog entry. Fashion fashion and mauris neque quam, fermentum ut nisl vitae, convallis maximus nisl. Sed mattis this blog entry. Fashion fashion and mauris neque quam, fermentum ut nisl vitae, convallis maximus nisl. Sed mattis
          </p>
          <p>
            More Hats! I am crazy about hats these days. Some text about this blog entry. Fashion fashion and mauris neque quam, fermentum ut nisl vitae, convallis maximus nisl. Sed mattis this blog entry. Fashion fashion and mauris neque quam, fermentum ut nisl vitae, convallis maximus nisl. Sed mattis
          </p>
        </div>
        <div
          class="displayFlexWith3columns "
        >
          <div
            class="displayFlexColumn"
          >
            <img
              alt="hat"
              src="avatar_girl2.jpg"
            />
            <div
              class="titleName"
            >
              <h5>
                My Name
              </h5>
              <h5>
                Just me, myself and I, exploring the universe of uknownment. I have a heart of love and a interest of lorem ipsum and mauris neque quam blog. I want to share my world with you.
              </h5>
            </div>
          </div>
          <div
            class="displayFlexColumn"
            style="padding-top: 1rem; margin: 0px; width: 100%;"
          >
            <div
              class="titleName"
            >
              <h5>
                My Popular post
              </h5>
            </div>
            <div
              class="displayFlex"
              style="padding: 5px;"
            >
              <div
                class="image"
              >
                <img
                  alt="avatar"
                  height="70"
                  src="avatar_smoke.jpg"
                  width="70"
                />
              </div>
              <div
                class="displayFlexColumn"
                style="margin-left: 1em; justify-content: space-evenly;"
              >
                <h5
                  style="margin: 0px;"
                >
                  Denim
                </h5>
                <h5
                  style="margin: 0px;"
                >
                  Sed mattis nunc
                </h5>
              </div>
            </div>
            <hr
              class="solid"
            />
            <div
              class="displayFlex"
              style="padding: 5px;"
            >
              <div
                class="image"
              >
                <img
                  alt="avatar"
                  height="70"
                  src="avatar_smoke.jpg"
                  width="70"
                />
              </div>
              <div
                class="displayFlexColumn"
                style="margin-left: 1em; justify-content: space-evenly;"
              >
                <h5
                  style="margin: 0px;"
                >
                  Denim
                </h5>
                <h5
                  style="margin: 0px;"
                >
                  Sed mattis nunc
                </h5>
              </div>
            </div>
            <hr
              class="solid"
            />
            <div
              class="displayFlex"
              style="padding: 5px;"
            >
              <div
                class="image"
              >
                <img
                  alt="avatar"
                  height="70"
                  src="avatar_smoke.jpg"
                  width="70"
                />
              </div>
              <div
                class="displayFlexColumn"
                style="margin-left: 1em; justify-content: space-evenly;"
              >
                <h5
                  style="margin: 0px;"
                >
                  Denim
                </h5>
                <h5
                  style="margin: 0px;"
                >
                  Sed mattis nunc
                </h5>
              </div>
            </div>
            <hr
              class="solid"
            />
          </div>
          <div
            class="displayFlexColumn"
            style="padding-top: 1rem; margin: 0px; width: 100%; height: 100%;"
          >
            <div
              class="titleName"
            >
              <h5>
                My Popular post
              </h5>
            </div>
            <div
              class="box"
              style="height: 100%; width: auto; background-color: grey; border: 15px solid #f1f1f1; padding: 30%; margin: 0px;"
            >
              <label
                style="display: flex; justify-content: center;"
              >
                Your ad here.
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
